includes/abstracts/abstract-wc-stripe-payment-gateway.php

```diff
@@ -1937,6 +1970,48 @@ public function javascript_params() {
 		return $stripe_params;
 	}
 
+	/**
+	 * Retrieves and returns the source_id for the given $_POST variables.
+	 *
+	 * @throws WC_Stripe_Exception Error while attempting to retrieve the source_id.
+	 * @return object
+	 */
+	private function get_source_object_from_request() {
+		if ( empty( $_POST['stripe_source'] ) && empty( $_POST['stripe_token'] ) ) {
+			throw new WC_Stripe_Exception( 'Missing stripe_source and stripe_token from the request.' );
+		}
+
+		$source = isset( $_POST['stripe_source'] ) ? wc_clean( wp_unslash( $_POST['stripe_source'] ) ) : '';
+
+		if ( ! empty( $source ) ) {
```

Comment:
Yes, the code here and below in this method is redundant.

[get_source_object](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/cbc6f7a84d163f8449a2dad5c0f98522048f89f6/includes/abstracts/abstract-wc-stripe-payment-gateway.php#L646-L667) doesn't handle $source_object being a WP_Error. In the current implementation, get_source_object would just return the WP_Error. Its docblock should be updated to reflect this.

A WP_Error is a possible return value from [WC_Stripe_API::retrieve](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/develop/includes/class-wc-stripe-api.php#L163-L188) -> [WC_Stripe_API::get_payment_method](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/develop/includes/class-wc-stripe-api.php#L276-L292).

I evaluated updating `get_source_object` to handle a WP_Error to avoid this redundancy, but I preferred to do it separately to keep the potential impact of this PR more minor.